### PR TITLE
Add helm chart.

### PR DESCRIPTION
**Pull request by chalko**
Basic helm file for url-redirector

### CODE REVIEW

File: helm/url-redirector/templates/deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        {{- toYaml .Values.podSecurityContext | nindent 8 }}
+      containers:
+        - name: {{ .Chart.Name }}
+          securityContext:
+            {{- toYaml .Values.securityContext | nindent 12 }}
+          image: "{{ .Values.image.repository }}:{{ .Values.image.tag | default .Chart.AppVersion }}"
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          ports:
+            - name: http
+              containerPort: 8080
```
Comment (BobyMCbobs) [inline helm/url-redirector/templates/deployment.yaml]: This should come from `.Values.service.port`

File: helm/url-redirector/templates/deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+          env:
+            {{- toYaml . | nindent 10 }}
+          {{- end }}
+          {{- with .Values.volumeMounts }}
+          volumeMounts:
+            {{- toYaml . | nindent 10 }}
+          {{- end }}
+          readinessProbe:
+            tcpSocket:
+              port: 8080
```
Comment (BobyMCbobs) [inline helm/url-redirector/templates/deployment.yaml]: This should come from `.Values.service.port`

File: helm/url-redirector/templates/deployment.yaml [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+            initialDelaySeconds: 2
+            periodSeconds: 10
+          livenessProbe:
+            tcpSocket:
+              port: 8080
```
Comment (BobyMCbobs) [inline helm/url-redirector/templates/deployment.yaml]: This should come from `.Values.service.port`

File: helm/url-redirector/values.yaml [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+      paths:
+        - path: /
+          pathType: ImplementationSpecific
+  tls: []
+  #  - secretName: chart-example-tls
+  #    hosts:
+  #      - chart-example.local
+
+
+env:
```
Comment (BobyMCbobs) [inline helm/url-redirector/values.yaml]: Could this be converted into extraEnv and the properties set below be set in the deployment directly?

File: helm/url-redirector/values.yaml [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+  - name: TZ
+    value: "Pacific/Auckland"
+  - name: APP_PORT
+    value: ":8080"
+  - name: APP_CONFIG_YAML
+    value: /app/cfg/config.yaml
+
+appConfig: |
```
Comment (BobyMCbobs) [inline helm/url-redirector/values.yaml]: nit: This would be nicer as just yaml without the string block (`|`)

**Comment by BobyMCbobs** [comment]
Thank you @chalko for your PR!
Glad to know that folks are picking up url-redirector :smiley: 

I've left some comments on some changes, cheers!

File: helm/url-redirector/templates/deployment.yaml (hunk 1 above)
Comment (chalko) [reply helm/url-redirector/templates/deployment.yaml]: done

File: helm/url-redirector/templates/deployment.yaml (hunk 2 above)
Comment (chalko) [reply helm/url-redirector/templates/deployment.yaml]: done

File: helm/url-redirector/templates/deployment.yaml (hunk 3 above)
Comment (chalko) [reply helm/url-redirector/templates/deployment.yaml]: done

File: helm/url-redirector/values.yaml (hunk 5 above)
Comment (chalko) [reply helm/url-redirector/values.yaml]: I can't figure out the yaml to make that work.  Lets work on that in a follow up change.

File: helm/url-redirector/values.yaml (hunk 4 above)
Comment (chalko) [reply helm/url-redirector/values.yaml]: This could be. I would prefer to also to this in a separate change

**Comment by chalko** [comment]
Ready for re-review.  I should not have sent the earlier comments until I updated the branch.  Still getting use to the flow here versus the gerrit flow I am used to with android.

File: helm/url-redirector/values.yaml (hunk 5 above)
Comment (BobyMCbobs) [reply helm/url-redirector/values.yaml]: No stress, sounds good!

File: helm/url-redirector/values.yaml (hunk 4 above)
Comment (BobyMCbobs) [reply helm/url-redirector/values.yaml]: All good, I'm also happy to implement it

File: helm/url-redirector/values.yaml [hunk 6]
```diff
@@ -0,0 +1,113 @@
+# Default values for url-redirector.
+# This is a YAML-formatted file.
+# Declare variables to be passed into your templates.
+
+replicaCount: 3
```
Comment (BobyMCbobs) [inline helm/url-redirector/values.yaml]: This should be set to just one replica

**Comment by BobyMCbobs** [comment]
> Ready for re-review. I should not have sent the earlier comments until I updated the branch. Still getting use to the flow here versus the gerrit flow I am used to with android.

Thank you, @chalko! I've left one more comment on a section.
Aside from that comment, may I ask the _helm/_ folder to be renamed to _charts/_?
With those changes we're ready for a squashing the commits and merge.

File: helm/url-redirector/values.yaml (hunk 6 above)
Comment (chalko) [reply helm/url-redirector/values.yaml]: done

**Comment by chalko** [comment]
Done

**Comment by chalko** [comment]
Should I squash locally and re-upload, or is this something you can do when you approve the request.

**Comment by BobyMCbobs** [comment]
Thank you for your changes, @chalko!
Usually squashing is done by the PR owner, but I just found a setting in GitHub that allows me to do this so i'll try it!

**Pull request merged**